src/NServiceBus.Core/Routing/RoutingOptionExtensions.cs

```diff
@@ -33,6 +33,18 @@ public static void OverrideReplyToAddressOfIncomingMessage(this ReplyOptions opt
         }
 
         /// <summary>
+        /// Returns the destination configured by <see cref="SetDestination(ReplyOptions, string)"/>.
+        /// </summary>
+        /// <param name="option">Option being extended.</param>
+        /// <returns>The specified destination address or <c>null</c> when no destination was specified.</returns>
+        public static string GetDestination(this ReplyOptions option)
+        {
+            UnicastReplyRouterConnector.State state;
```

Comment:
hm we don't have any guards on option in the `RoutingOptiionExtensions` but we have on `CorrelationContextExtensions`. Going to add the guards